xyz-psql-connector/src/main/java/com/here/xyz/psql/PSQLXyzConnector.java

```diff
@@ -129,38 +129,73 @@ protected XyzResponse processGetFeaturesByBBoxEvent(GetFeaturesByBBoxEvent event
 
       final BBox bbox = event.getBbox();
 
-      if(event.getTweakType() != null)
-      { Map<String, Object> tweakParams = event.getTweakParams();
+      boolean bTweaks = ( event.getTweakType() != null ),
+              bOptViz = "viz".equals( event.getOptimizationMode() );
 
-        switch (event.getTweakType().toLowerCase()) {
+      if( !bOptViz && event.getSelection() != null && "*".equals( event.getSelection().get(0) ) ) //correction for raw usecase
+       event.setSelection(null);
+
+      if( bTweaks || bOptViz )
+      { String tweakType;
+        Map<String, Object> tweakParams;
+        boolean bVizSamplingOff = false;
+
+        if( bTweaks )
+        { tweakType   = event.getTweakType().toLowerCase();
+          tweakParams = event.getTweakParams();
+        }
+        else
+        { tweakType   = TweaksSQL.ENSURE;
+          event.setTweakType( TweaksSQL.ENSURE );
+          tweakParams = new HashMap<String, Object>();
+          switch( event.getVizSampling().toLowerCase() )
```

Comment:
Possible NPE when event.getVizSampling() is null

Reply:
event.getVizSampling() is always set. defaults to "med"